aws/security_hub.go

```diff
@@ -0,0 +1,182 @@
+package aws
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/securityhub"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/report"
+	"github.com/gruntwork-io/cloud-nuke/telemetry"
+	"github.com/gruntwork-io/go-commons/errors"
+	commonTelemetry "github.com/gruntwork-io/go-commons/telemetry"
+)
+
+func getAllSecurityHubArns(session *session.Session, excludeAfter time.Time) ([]string, error) {
+	svc := securityhub.New(session)
+	var securityHubArns []string
+
+	output, err := svc.DescribeHub(&securityhub.DescribeHubInput{})
+
+	if err != nil {
+		// If Security Hub is not enabled when we call DescribeHub, we get back an error
+		// so we should ignore the error if it's just telling us the account/region is not
+		// subscribed and return nil to indicate there are no resources to nuke
+		if strings.Contains(err.Error(), "is not subscribed to AWS Security Hub") {
+			return nil, nil
+		}
+		return nil, errors.WithStackTrace(err)
+	}
+
+	if shouldIncludeHub(output, excludeAfter) {
+		securityHubArns = append(securityHubArns, *output.HubArn)
+	}
+
+	return securityHubArns, nil
+}
+
+func shouldIncludeHub(hub *securityhub.DescribeHubOutput, excludeAfter time.Time) bool {
+	subscribedAt, _ := time.Parse(time.RFC3339, *hub.SubscribedAt)
```

Comment:
We definitely want to handle the error returned by `time.Parse` - otherwise this could be confusing to debug in the future.